src/main/java/com/google/sps/data/JobJSON.java

```diff
@@ -79,4 +81,128 @@ public JobJSON(String projectId, String name, String id, String type, String sdk
     this.price = price;
     this.sdkName = sdkName;
   }
+  
+  // Helper function for the equals method
+  // Returns true if the 2 fields are different
+  private boolean notEqualsField(Double f1, Double f2) {
+    if ((f1 == null && f2 != null) ||
+        (f1 != null && f2 == null) ||
+        (f1 != null && f2 != null && f1.compareTo(f2) != 0)) {
+      return true;
+    }
+
+    return false;
+  }
+  
+  // Helper function for the equals method
+  // Returns true if the 2 fields are different
+  private boolean notEqualsField(Integer f1, Integer f2) {
+    if ((f1 == null && f2 != null) ||
+        (f1 != null && f2 == null) ||
+        (f1 != null && f2 != null && f1.compareTo(f2) != 0)) {
+      return true;
+    }
+
+    return false;
+  }
+  
+  // Helper function for the equals method
+  // Returns true if the 2 fields are different
+  private boolean notEqualsField(String f1, String f2) {
+    if ((f1 == null && f2 != null) ||
+        (f1 != null && f2 == null) ||
+        (f1 != null && f2 != null && f1.compareTo(f2) != 0)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof JobJSON)) {
+      return false;
+    }
+    JobJSON job2 = (JobJSON) obj;
+
+    if (notEqualsField(this.id, job2.id)) {
+      return false;
+    }
+    if (notEqualsField(this.projectId, job2.projectId)) {
+      return false;
+    }
+    if (notEqualsField(this.name, job2.name )) {
+      return false;
+    }
+    if (notEqualsField(this.type , job2.type)) {
+      return false;
+    }
+    if (notEqualsField(this.sdk, job2.sdk)) {
+      return false;
+    }
+    if (notEqualsField(this.sdkSupportStatus, job2.sdkSupportStatus)) {
+      return false;
+    }
+    if (notEqualsField(this.region, job2.region)) {
+      return false;
+    }
+    if (this.currentWorkers != job2.currentWorkers) {
+      return false;
+    }
+    if (notEqualsField(this.startTime, job2.startTime)) {
+      return false;
+    }
+    if (notEqualsField(this.totalVCPUTime, job2.totalVCPUTime)) {
+      return false;
+    }
+    if (notEqualsField(this.totalMemoryTime, job2.totalMemoryTime)) {
+      return false;
+    }
+    if (notEqualsField(this.totalDiskTimeHDD, job2.totalDiskTimeHDD)) {
+      return false;
+    }
+    if (notEqualsField(this.totalDiskTimeSSD, job2.totalDiskTimeSSD)) {
+      return false;
+    }
+    if (notEqualsField(this.currentVcpuCount, job2.currentVcpuCount)) {
+      return false;
+    }
+    if (this.enableStreamingEngine != job2.enableStreamingEngine) {
```

Comment:
and here? Is this because it's a Boolean?

Reply:
The same applies to boolean, which is a primitive like int.